btree/btree.c

```diff
@@ -3085,6 +3085,594 @@ static void ff_rec_del_credit(const struct nd *node, m0_bcount_t ksize,
+ *  Section START -
+ *  Fix Sized Keys and Variable Sized Value Node Structure
+ *  ---------------------------------------------------------
+ */
+
+/**
+ *
+ * Proposed Node Structure for Fix Sized Keys and Variable Sized Value :
+ *
+ * +-----------+----+-+----+-+----+-+------------------------------------+----+----+----+
+ * |           |    | |    | |    | |                                    |    |    |    |
+ * |           |    | |    | |    | |                                    |    |    |    |
+ * |Node Header| K0 | | K1 | | K2 | | ----->                      <----- | V2 | V1 | V0 |
+ * |           |    | |    | |    | |                                    |    |    |    |
+ * |           |    | |    | |    | |                                    |    |    |    |
+ * +-----------+----+++----+++----+++------------------------------------+----+----+----+
+ *                   |      |      |                                     ^    ^    ^
+ *                   |      |      |                                     |    |    |
+ *                   |      |      |                                     |    |    |
+ *                   +------+------+-------------------------------------+    |    |
+ *                          |      |                                          |    |
+ *                          +------+------------------------------------------+    |
+ *                                 |                                               |
+ *                                 +-----------------------------------------------+
+ *
+ * The above structure represents the way fixed key size and variable value size
+ * node format will get stored in memory.
+ *
+ * Node can be mainly divided into three region, node header, region for keys
+ * and region for values.
+ *
+ * Node header will contain all the relevant information about the node
+ * including node type.
+ *
+ * The key region will be start after the node header region. Keys will be added
+ * (from the end of node header region) in the sorted order and each key will be
+ * associated with offset which will be the offset for the value. The offset for
+ * value will be useful to get absolute address of value associated with that
+ * key.
+ *
+ * The values will be added from the end of the node such that value for the
+ * first record will be present at the end of the node. To get the starting
+ * address of value at index i, we will add offset associated with ith key to
+ * the end address of the node; the size of the value will be difference of
+ * offset for value at index i and offset for previous value.
+ *
+ * In this way, keys (which will also include offset for value) will be added
+ * from right to left starting after node header region and value will be added
+ * from left to right starting from end of the node.
+ *
+ */
+struct fkvv_head {
+	struct m0_format_header  fkvv_fmt;    /*< Node Header */
+	struct node_header       fkvv_seg;    /*< Node type information */
```

Comment:
Kindly add this comment as it has been added to ff and vkvv head structures.
	`
	/**
	 * The above 2 structures should always be together with node_header
	 * following the m0_format_header.
	 */
`